src/OrleansAzureUtils/Providers/Lease/AzureBlobLeaseProvider.cs

```diff
@@ -0,0 +1,140 @@
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.LeaseProviders
+{
+    public class AzureBlobLeaseProviderConfig
+    {
+        public string DataConnectionString { get; set; }
+        public string BlobContainerName { get; set; }
+    }
+    public class AzureBlobLeaseProvider : ILeaseProvider
+    {
+        private CloudBlobContainer container;
+        private AzureBlobLeaseProviderConfig providerConfig;
+        private CloudBlobClient blobClient;
+        public AzureBlobLeaseProvider(AzureBlobLeaseProviderConfig config)
+        {
+            var account = CloudStorageAccount.Parse(config.DataConnectionString);
+            this.blobClient = account.CreateCloudBlobClient();
+            this.providerConfig = config;
+        }
+
+        private async Task InitContainerIfNotExistsAsync()
+        {
+            if (this.container == null)
+            {
+                this.container = blobClient.GetContainerReference(this.providerConfig.BlobContainerName);
+                await this.container.CreateIfNotExistsAsync().ConfigureAwait(false); 
+            }
+        }
+
+        public async Task<AcquireLeaseResult[]> Acquire(string category, LeaseRequest[] leaseRequests)
+        {
+            await InitContainerIfNotExistsAsync();
+            var tasks = new List<Task<AcquireLeaseResult>>();
+            foreach (var leaseRequest in leaseRequests)
+            {
+                tasks.Add(Acquire(category, leaseRequest));
+            }
+            //Task.WhenAll will return results for each task in an array, in the same order of supplied tasks
+            return await Task.WhenAll(tasks);
+        }
+
+        private string GetBlobName(string category, string resourceKey)
+        {
+            return $"{category.ToLower()}-{resourceKey.ToLower()}.json";
+        }
+
+        private async Task<AcquireLeaseResult> Acquire(string category, LeaseRequest leaseRequest)
+        {
+            try
+            {
+                var blob = this.container.GetBlockBlobReference(GetBlobName(category, leaseRequest.ResourceKey));
+                blob.Properties.ContentType = "application/json";
+                //create this blob
+                await blob.UploadTextAsync("blob");
+                var leaseId = await blob.AcquireLeaseAsync(leaseRequest.Duration);
+                return new AcquireLeaseResult(new AcquiredLease(leaseRequest.ResourceKey, leaseRequest.Duration, leaseId, DateTime.UtcNow), ResponseCode.OK, null);
+            }
+            catch (StorageException e)
+            {
+                ResponseCode statusCode;
+                //This mapping is based on references : https://docs.microsoft.com/en-us/rest/api/storageservices/blob-service-error-codes
+                // https://docs.microsoft.com/en-us/rest/api/storageservices/Lease-Blob?redirectedfrom=MSDN
+                switch (e.RequestInformation.HttpStatusCode)
+                {
+                    case 404:
```

Comment:
We can use HttpStatusCode enum. But since it is an enum and `e.RequestInformation.HttpStatusCode` is an int , the code need to parse enum to int to use in switch block or using a "if else" block rather than switch, both of which are more performance heavier. 

And azure API e.RequestInformation.HttpStatusCode is an int. So we can say we are following azure standard. :)